### PR TITLE
Compression 2021 chapter contents

### PR DESCRIPTION
**Pull request by lvandeve**
#2160

### CODE REVIEW

File: src/content/en/2021/compression.md [hunk 1]
```diff
@@ -1,20 +1,375 @@
 ---
 #See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
 title: Compression
-description: TODO
+description: Compression chapter of the 2021 Web Almanac covering HTTP compression, algorithms, compression levels, content types, 1st party and 3rd party compression and opportunities.
 authors: [lvandeve, mo271, jyrkialakuijala]
```
Comment (rviscomi) [inline src/content/en/2021/compression.md:5]: @lvandeve @mo271 @jyrkialakuijala could each of you add a bio here to the metadata in the form `<author_id>_bio: Foo`? See the [wiki](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) for more info.

@mo271 @jyrkialakuijala you could reuse bios from [last year](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/content/en/2020/compression.md?plain=1#L10-L12) if you'd like.

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: src/content/en/2021/compression.md (hunk 1 above)
Comment (lvandeve) [reply src/content/en/2021/compression.md:5]: Done, added @mo271 @jyrkialakuijala from last year and a new one for @lvandeve

**Comment by lvandeve** [comment]
Added the bios, PTAL

**Comment by lvandeve** [comment]
I pushed another update to add the png images generated by the script, fix a few image links, and updated the style of the compression level images to match the style of the other charts better

**Review by rviscomi — APPROVED** [review]
LGTM. @bazzadp any comments before we merge?

**Comment by tunetheweb** [comment]
We're missing a screenshot for this:

```
{{ figure_markup(
  image="content-encoding.png",
  caption='Use DevTools to check if content encoding is used on your site',
  description="Image showing how to use DevTools to see if content encoding is used.",
  width=591,
  height=939
  )
}}
```

I presume the intent was to reuse the 2020 image for this so will fix the markup to do that.

Also going to fix some minor issues with the graphs (font sizes, labels and inconsistent decimal places) and will add to this PR before we merge so copy editing can concentrate just on the copy.

**Comment by lvandeve** [comment]
> I presume the intent was to reuse the 2020 image for this so will fix the markup to do that.

Indeed it was, thank you for fixing this

**Comment by tunetheweb** [comment]
It is disappointing how much of this content is a straight copy from the 2020 chapter which was already a light chapter.

Most of the chapter (including the opening and the conclusion) is word for word identical to the 2020 chapter. While the new insights on compression levels are very interesting, I do not think the rest of the chapter is sufficiently different to warrant publishing this as part of the 2021 Web Almanac. To do so would reflect badly on the other chapters who have made significant efforts to review, rewrite and reinterpret their topics to give a fresh chapter for this year.

Therefore I am not going to merge this, and given our launch date on this Wednesday the 1st December, I will unfortunately be left with no choice but to close this PR and remove the chapter from the 2021 edition, if the chapter is not significantly differentiated from the 2020 edition by the end of tomorrow (Monday 29th November). Please let me know if you intend to rewrite the duplicated content by then.

**Comment by lvandeve** [comment]
Yes, I will differentiate it. The same (but updated) data as last year is used for many sections to allow comparing with last year. The section about compression levels is new and significantly different.

@tunetheweb please let me know if rewriting the word for word sections, but keeping the content they describe the same (that is, describe similar data as last year but with the updated numbers), would be acceptable

**Comment by tunetheweb** [comment]
Yes reviewing and rewriting the identical content, even if the message stays largely the same is acceptable (assuming that is what the data is telling you). Straight copying and pasting with only minor tweaks is not.

**Review by rviscomi — CHANGES_REQUESTED** [review]
Thanks @lvandeve. I hadn't realized that the content was so close to 2020's chapter, so I agree with Barry's suggestion to rewrite it to be more unique this year. If completing it by tomorrow isn't feasible let's find a timeframe that works for everyone.

**Comment by lvandeve** [comment]
Sorry about that.

I have re-worded the sections "introduction", "how to analyze compression on your site" and "conclusion" and the first paragraph of "What type of content should we compress?". I also replaced the chrome developer tools in the analyze section with a new one.

Still to re-word are: parts of "how to use http compression", "first party vs third party compression", and a few sentences of "How to improve on compression". I'll also ask several people to review the updated text.

"Compression levels" and most of "How to improve on compression" are new and so don't need this, and "Current state of HTTP compression" was also already very different

**Comment by lvandeve** [comment]
Went through the text together with Moritz and changed the emphasis of the affected sections, none of the text is a copy of last year anymore

**Comment by tunetheweb** [comment]
Thanks you very much for actioning this so quickly @lvandeve . It looks a lot better now and we're good to merge. Just a quick copy edit after in a new PR and we'll be done.

**Review by tunetheweb — APPROVED** [review]

File: src/content/en/2021/compression.md [hunk 2]
```diff
@@ -1,20 +1,271 @@
...
 reviewers: [fischbacher, eustas]
 analysts: [paulcalvano]
 editors: []
 translators: []
+jyrkialakuijala_bio: Jyrki Alakuijala is an active member of the open source software community, and a data compression researcher. Jyrki works at Google as a Technical Lead/Manager, and his recent published work has been with Zopfli, Butteraugli, Guetzli, Gipfeli, WebP lossless, Brotli, and JPEG XL compression formats and algorithms, and two hashing algorithms, CityHash, and HighwayHash. Before his Google employment he developed software for neurosurgery and radiation therapy treatment planning.
+mo271_bio: Moritz Firsching is software engineer at Google Switzerland, where he works on progressive image formats and font compression. Before that Moritz did research as a mathematician studying polytopes.
```
Comment (fischbacher) [inline src/content/en/2021/compression.md:11]: Have you checked with Moritz that this is a good description? I would say he's done (and is still doing) quite a few other things as well in mathematics.

File: src/content/en/2021/compression.md [hunk 3]
```diff
@@ -1,20 +1,271 @@
...
+featured_stat_label_2: Compressed responses which use Gzip
+featured_stat_3: 72%
+featured_stat_label_3: Websites that pass the Lighthouse audit with maximum score on text compression
+unedited: true
 ---
 
-## TODO
+## Introduction
+
+When loading websites it is important to make the best use of the available bandwidth. HTTP allows to compress the content that it delivers and this decreases the time needed to transfer the content. Even when taking the time to compress and decompress the content into account, this often leads to significant improvement in the user experience, it can reduce [page weight](./page-weight), improve [web performance](./performance) and boost search rankings, so it's an important part of [search engine optimization](./seo). On the other hand, not using compression, or using it with a suboptimal configuration, can result in longer wait times for the user and can negatively affect the growth of a web service.
```
Comment (fischbacher) [inline src/content/en/2021/compression.md:26]: Missing comma: "When loading websites, it is ..."

File: src/content/en/2021/compression.md [hunk 4]
```diff
@@ -1,20 +1,271 @@
...
+  caption='Chrome DevTools checking the content-encoding of responses',
+  description="Image showing how to use Chrome DevTools to see if content encoding is used.",
+  width=591,
+  height=939
+  )
+}}
+
+ A different tool that can analyze compression on your site is Google's <a hreflang="en" href="https://developers.google.com/web/tools/lighthouse">Lighthouse</a> tool. It runs a series of audits, including the <a hreflang="en" href="https://web.dev/uses-text-compression/">"Enable text compression" audit</a>. This audit attemps to compress resources to check if they reduced by at least 10% and 1,400 bytes. Depending on the score, it can show a compression recommendation in the results, with a list of the resources that can be compressed to benefit your website.
+
+The HTTP Archive [runs Lighthouse audits](./methodology#lighthouse) for each mobile page, and from this data we can see that 72% of websites pass this audit. This is 2% less than [last year's](../2020/compression#identifying-compression-opportunities) 74% which is, despite more usage of text compression overall compared to last year, a slight drop.
```
Comment (fischbacher) [inline src/content/en/2021/compression.md:213]: s/for each/for every/ - but technically, both are permissible here.

File: src/content/en/2021/compression.md [hunk 5]
```diff
@@ -1,20 +1,271 @@
...
+
+## Conclusion
+
+The analysis on compression levels reveals that about 6% of Gzip-compressed content uses more advanced compressors such as Zopfli, while a similar \'optimal parsing\' approach is used for 17% of Brotli-compressed content. This indicates that when much slower but slightly more efficient methods are available, a significant amount of users will deploy these methods for their static content such as JavaScript.
+
+Usage of HTTP compression continous to grow, and especially Brotli has increased significantly compared to the [previous year's Almanac](../2020/compression). The number of HTTP responses using any text compression increased 2%, while Brotli increased over 4%. Despite the increase, we still see opportunities to use more HTTP compression by tweaking compression settings of servers. You can benefit from taking a closer look at your own website's responses and your server configuration for the compression methods and their level. Tuning the default compression settings for the most popular HTTP servers could have a great impact for users.
```
Comment (fischbacher) [inline src/content/en/2021/compression.md:268]: Typo: s/continuous/continues/

Comment (fischbacher) [inline src/content/en/2021/compression.md:268]: "increased x%" is ambiguous. It's better to say something like "increased by so-and-so-many percentage points".

**Review by fischbacher — COMMENTED** [review]
First pass over the text.

File: src/content/en/2021/compression.md (hunk 2 above)
Comment (lvandeve) [reply src/content/en/2021/compression.md:11]: Yes, Moritz is happy with the bio

**Pull request merged**